### PR TITLE
FireFlies

### PR DESCRIPTION
**Pull request by Duckjr94**
A module that allows the creation of glass jars that, at night, open up and create “firefly” particles around your build

### CODE REVIEW

**Comment by Bloo-dev** [comment]
Can we get some screenshots? :)

**Comment by Kroppeb** [comment]
The folder should be called `gm4_fireflies`

**Comment by Duckjr94** [comment]
> Can we get some screenshots? :)

Sure! I meant to add the datapack files as well, this is my first time on github haha
943FD75D-F95D-48FF-9093-340289DD2D47

**Comment by sparks0** [comment]
I'm a bit confused. I don't see any code for getting firefly glass bottles. How does one create a firefly jar?

File: fireflies/data/fireflies/functions/init.mcfunction [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#announce module installation
+tellraw @a[gamemode=creative] ["",{"text":"[GM4]: Installing FireFlies..."}]
+execute unless entity @p run say GM4: Installing FireFlies...
```
Comment (AndanteDevs) [inline fireflies/data/fireflies/functions/init.mcfunction]: Needs to be updated to the latest format - look at other data packs for a reference

File: fireflies/data/fireflies/functions/init.mcfunction [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+tellraw @a[gamemode=creative] ["",{"text":"[GM4]: Installing FireFlies..."}]
+execute unless entity @p run say GM4: Installing FireFlies...
+
+#declare and initialise scoreboards and settings
+scoreboard players set update_happened gm4_up_check 1
+scoreboard players set fireflies gm4_modules 1
+scoreboard players set fireflies gm4_clock_tick 0
+
+# create scoreboard
+scoreboard objectives add gm4_fireflies dummy "gm4_fireflies"
```
Comment (AndanteDevs) [inline fireflies/data/fireflies/functions/init.mcfunction]: GM4 generally doesn't use the display field

File: fireflies/pack.mcmeta [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  "pack":{
+    "pack_format":0,
+    "description":"A Gamemode 4 Module"
+  },
+  "module_name":"FireFlies",
+  "module_id":"fireflies",
+  "site_description":"Want to spice up your builds? Craft a bottle of FireFlies and watch them roam in the night!",
+  "site_categories":[
+  ],
+  "video_link":"https://www.youtube.com/",
```
Comment (AndanteDevs) [inline fireflies/pack.mcmeta]: Can be left blank

File: fireflies/data/fireflies/functions/destroy.mcfunction [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#checks for slab on top
+#called from main
+
+execute as @s run fill ~ ~ ~ ~ ~ ~ minecraft:air destroy
```
Comment (AndanteDevs) [inline fireflies/data/fireflies/functions/destroy.mcfunction]: Can't you do `setblock ~ ~ ~ air destroy`?

**Comment by Kroppeb** [comment]
> I'm a bit confused. I don't see any code for getting firefly glass bottles. How does one create a firefly jar?

You create a bottle by dropping a bottle named “Fireflies” and a piece of glow stone IIRC. But it seems a file is misssing

**Comment by sparks0** [comment]
I like the module but I don't think I like that as a method for getting firefly bottles @Kroppeb

**Pull request closed without merging**